### DISCUSSION TITLE
Is there a status page for the Renovate Bot?

### DISCUSSION THREAD

**Discussion by Leksat** [renovatebot/renovate#17632] [Request Help] [answered]
Currently I'm getting timeouts on my repository:

<img width="747" alt="Screen Shot 2022-09-05 at 14 35 41" src="https://user-images.githubusercontent.com/989015/188440193-19e50bbc-7136-4f6c-b56f-2b09da7a6b00.png">

The `timeout` logs end with:

<img width="671" alt="Screen Shot 2022-09-05 at 14 37 01" src="https://user-images.githubusercontent.com/989015/188440243-88b8ac81-2b38-4825-baef-ce80c20c9188.png">

Is this a temporary problem with the Renovate Bot? Or maybe the recent changes on the [repository](https://github.com/AmazeeLabs/silverback-mono) caused this?

Unfortunatelly I could not google something like https://status.renovatebot.com/ to answer the question myself.

**Comment by rarkins** [comment, marked as answer]
There's no status page currently, but if you have timeouts persisting for as long as that then it's not a temporary problem.

**Reply by Leksat** [reply]
Hmm... Is there a way to debug my case? 

The log is full of `"check": "stabilityDays"` - could it be the cause of the slowness? Maybe I should try removing it?

**Comment by rarkins** [comment]
With timestamps:

```
{"level":30,"renovateVersion":"32.189.0","msg":"Repository started","time":"2022-09-05T10:18:20.125Z"}
```

extraction was complete 13s later:

```
{"level":30,"baseBranch":"development","stats":{"managers":{"composer":{"fileCount":12,"depCount":58},"docker-compose":{"fileCount":1,"depCount":2},"github-actions":{"fileCount":6,"depCount":27},"npm":{"fileCount":54,"depCount":652}},"total":{"fileCount":73,"depCount":739}},"msg":"Dependency extraction complete","time":"2022-09-05T10:18:33.356Z"}
```

It appeared to hang during lookup:

```
{"level":20,"dependency":"react-rxjs","msg":"Found deprecationMessage","time":"2022-09-05T10:20:28.977Z"}
{"level":30,"msg":"SIGTERM signal received for Renovate App","time":"2022-09-05T11:04:43.047Z"}
```

There's definitely something strange/slow about the lookups, but I'm not sure if stabilityDays is necessarily it. It could be network contention from too many parallel lookups.

**Reply by Leksat** [reply]
The next jobs worked well. I did no changes to the repo. So yeah most probably it was some temporary network issue 🤷

**Reply by Leksat** [reply]
@rarkins Timeouts are back 😢

<img width="870" alt="Screen Shot 2022-09-08 at 17 39 57" src="https://user-images.githubusercontent.com/989015/189151522-440b2ccc-0aa2-40ef-a320-d46c47989b89.png">

The last one:  https://app.renovatebot.com/dashboard#github/AmazeeLabs/silverback-mono/805084708

Is there any chance to debug it myself? Could you recommend something in general?

**Reply by stefanmo** [reply]
I have the same issue since three days ago and no idea how to debug it 😞

**Reply by Leksat** [reply]
@stefanmo do you use the `stabilityDays` feature?

**Reply by rarkins** [reply]
The common theme for these timeouts seems to be the npm registry. Example:

```
{
  "level": 20,
  "err": {
    "name": "AggregateError",
    "message": "
    Error: external-host-error
        at getDependency (/home/ubuntu/renovateapp/node_modules/renovate/dist/modules/datasource/npm/get.js:164:19)
        at runMicrotasks (<anonymous>)
        at async NpmDatasource.getReleases (/home/ubuntu/renovateapp/node_modules/renovate/dist/modules/datasource/npm/index.js:26:21)
        at async getRegistryReleases (/home/ubuntu/renovateapp/node_modules/renovate/dist/modules/datasource/index.js:57:17)
        at async fetchReleases (/home/ubuntu/renovateapp/node_modules/renovate/dist/modules/datasource/index.js:240:23)",
    "stack": "AggregateError: 
    Error: external-host-error
        at getDependency (/home/ubuntu/renovateapp/node_modules/renovate/dist/modules/datasource/npm/get.js:164:19)
        at runMicrotasks (<anonymous>)
        at async NpmDatasource.getReleases (/home/ubuntu/renovateapp/node_modules/renovate/dist/modules/datasource/npm/index.js:26:21)
        at async getRegistryReleases (/home/ubuntu/renovateapp/node_modules/renovate/dist/modules/datasource/index.js:57:17)
        at async fetchReleases (/home/ubuntu/renovateapp/node_modules/renovate/dist/modules/datasource/index.js:240:23)
    at next (/home/ubuntu/renovateapp/node_modules/p-map/index.js:43:14)
    at /home/ubuntu/renovateapp/node_modules/p-map/index.js:59:6
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)"
  },
  "msg": "Aggregate error is thrown",
  "time": "2022-09-09T05:13:07.698Z"
}
```

Once that happens, it should ideally be exiting with external host error although right now it seems like it's still waiting on some promises and eventually times out instead. Either way it's the npm registry error which is the root cause, which we're trying to work out.

Likely it's the same root cause as https://github.com/renovatebot/renovate/issues/17677 so better to switch this conversation to there. I will lock this one so that we can unify the discussion

**Comment by rarkins** [comment]
Actually let's treat this discussion as answered. There's no status page, nor would it help in a case like this

**Reply by Leksat** [reply]
Could not find a way to do it until I changed the discussion type to `Questions & Answers` 😅
